test/integration/signup_logon_test.js

```diff
@@ -0,0 +1,86 @@
+var system = require('system');
+var user = system.env.BT_TEST_MAIN_USER;
+var pass = system.env.BT_TEST_MAIN_PASS;
+
+var host = 'http://localhost:3000';
+
+function checkBoxes() {
+  return [
+    document.querySelector('#block_new_accounts').checked,
+    document.querySelector('#block_low_followers').checked,
+    document.querySelector('#share_blocks').checked,
+    document.querySelector('#follow_blocktogether').checked
+  ];
+}
+
+// API docs: http://docs.casperjs.org/en/latest/testing.html
+//
+// The first argument is the name of the test, the second is the
+// number of assertions occuring in the test, and the last is the test
+// block itself.
+casper.test.begin('Sign up and log on', 5, function(test) {
+  casper.start(host, function() {
+    return this.fill('form[action*="/auth/twitter"]', {}, true);
+  });
+
+  casper.then(function() {
+    return this.fill(
+      'form[action*="https://api.twitter.com/oauth/authenticate"]',
+      // NB: must use single quotes
+      { 'session[username_or_email]': user , 'session[password]': pass }, true);
+  });
+
+  casper.waitForSelector('.container-fluid', function() {
+    var checks = this.evaluate(checkBoxes);
+
+    test.assertEqual(checks, [false, false, false, true], 'new account has default settings');
+
+    this.click('#block_new_accounts');
+
+    this.reload(function() {
+      var checks = this.evaluate(checkBoxes);
+      test.assertEqual(checks, [true, false, false, true], 'block_new_accounts was saved');
+    });
+  });
+
+  casper.then(function() {
+    casper.open(host + '/logout', function() {
```

Comment:
I used `casper.open` here instead of `this.click('.logout')` since for some reason the latter didn't work.